demo/src/main/java/com/example/demo/MyController.java

```diff
@@ -1,48 +1,130 @@
-        if (!users.containsKey(username)) {
+    public ResponseEntity<String> login(@RequestParam("username") String username,
+                                        @RequestParam("password") String password) {
+        String token = userService.checkUser(username, password);
+        if (token == null) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+        return ResponseEntity.ok(token);
+    }
+
+    @PostMapping("/register")
+    public ResponseEntity<String> register(@RequestBody ObjectNode objectNode) {
+        String username = objectNode.get("name").asText();
+        String password = objectNode.get("pass").asText();
+        if (userService.isUsernameTaken(username)) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).build();
+        }
+        userService.generateUser(username, password);
+        return ResponseEntity.ok("User registered successfully");
+    }
+    @GetMapping("/hello")
+    public ResponseEntity<String> helloWorld() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication == null || !(authentication.getPrincipal() instanceof UserDataBase)) {
             return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
         }
-        if (!(users.get(username).equals(password))) {
+        return ResponseEntity.ok("Hello, world");
+    }
+
+    @PostMapping("/upload")
+    public ResponseEntity<String> uploadVideo(
+            @RequestParam("file") MultipartFile file,
+            @RequestHeader("Authorization") String token) throws IOException, NoSuchAlgorithmException {
+        String id = videoService.generateVideo(file, token);
+        if (id == null) {
             return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
         }
-        String token;
-        if (usersWhoHaveToken.containsKey(username)) {
-            token = usersWhoHaveToken.get(username).getToken();
+        return ResponseEntity.ok(id);
+    }
+
+    @GetMapping("/download")
+    public ResponseEntity<byte[]> downloadVideo(@RequestParam("Id") String id,
+                                                @RequestHeader("Authorization") String token) {
+        byte[] data = videoRepository.findVideoByIdentifier(id).getData();
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_OCTET_STREAM);
+        headers.setContentDisposition(ContentDisposition.builder("inline").
+                filename("video.mp4").build());
+
+        return new ResponseEntity<>(data, headers, HttpStatus.OK);
+    }
+
+    @GetMapping("/sortVideo")
+    public ResponseEntity<String> getTopVideo() {
+        videoRepository.updateVideos();
+        return ResponseEntity.ok("success");
+    }
+
+    @GetMapping("/like")
+    public String addLike(@RequestParam("Id") String id,
```

Comment:
It is better to nearly always return JSON (so some object instead of a String).
Or return ResponseEntity with HTTP OK code on success and HTTP Bad Request if like already exists